src/types.rs

```diff
@@ -0,0 +1,139 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// Magic bytes used to signal the start of the TEE attestation process.
+pub const MAGIC_BYTES: &[u8; 6] = b"TEETLS";
+
+/// Label used for exporting key material in the TEE attestation process.
+pub const EKM_LABEL: &[u8; 21] = b"EXPORTER-pluto-notary";
+
+/// Context used for exporting key material in the TEE attestation process.
+pub const EKM_CONTEXT: &[u8; 3] = b"tee";
+
+/// A struct representing a dummy token used for testing purposes.
+///
+/// The `DummyToken` struct is used to generate TEE attestation tokens by providing a predefined token value.
+/// It contains a single field, `body`, which holds the token value as a string.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct DummyToken {
+    /// The token value that the generator will use to create the TEE attestation token.
+    pub body: String,
+}
+
+/// A struct representing a JSON Web Token (JWT) used for TEE attestation.
+///
+/// The `JwtToken` struct contains various fields that represent the claims and attributes
+/// of a JWT used in the TEE attestation process.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct JwtToken {
```

Comment:
Is this not present in some crate that we already have? I feel like it would be. No big deal if not. Could just remove some code (though it may make us make a wrapper struct anyway)

This is kinda a nit.

Reply:
I doubt we'd have this particular JWT structure somewhere. It's Google's JWT for a TEE attestation.